### PR TITLE
CAMEL-20008 - Improve assertion to compare xml and not string in XMLTokenExpressionIteratorTest

### PR DESCRIPTION
**Pull request by apupier**
* With Java 21, the order of attributes are different but it does not change the semantic
* factorize the xml-unit version in parent pom

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -2697,6 +2697,18 @@
                 <type>pom</type>
                 <scope>import</scope>
             </dependency>
+            <dependency>
```
Comment (oscerd) [inline parent/pom.xml:2700]: Unless there is a good reason I would avoid adding other dependencies in parent/pom

**Review by oscerd — CHANGES_REQUESTED** [review]
just one minor thing.

File: parent/pom.xml (hunk 1 above)
Comment (apupier) [reply parent/pom.xml:2700]: for `xmlunit-core`, it was already used in several projects.

for `xmlunit-assertj3`, it allows to write simpler code and have a better error message

Comment (apupier) [reply parent/pom.xml:2700]: I can update to use the same xmlunit-core in this PR, and provide a different PR to migrate from xmlunit-core to xmlunit-assertj3 to separate concerns and discuss if it is worthy or not to use the introduced one.

Comment (apupier) [reply parent/pom.xml:2700]: Second PR provided to replace unitcore with assertj https://github.com/apache/camel/pull/11774

Comment (apupier) [reply parent/pom.xml]: the 2 other PRs have been merged.
Less code change in this PR

**Review by davsclaus — APPROVED** [review]

**Pull request merged**